### PR TITLE
Add nodelet to driver

### PR DESCRIPTION
**Pull request by georgflick**
Changes:
- Adding nodelet to the driver

Test together with PR in main repo.

### CODE REVIEW

File: CMakeLists.txt
Comment (Roboterbastler) [inline CMakeLists.txt:1]: I haven't tested it yet but I think usually the nodelet file also needs to be installed by:

```cmake
install(FILES nodelets.xml DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
```

Comment (georgflick) [reply CMakeLists.txt:1]: true, I added that now in the commit:
https://github.com/enwaytech/livox_ros_driver2/pull/6/commits/a69afe4f9e450f5fdfc38d1a6d432c59e763cfc8

File: include/livox_ros_driver2/nodelet.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class Nodelet : public nodelet::Nodelet
+{
+public:
+  Nodelet() = default;
+  ~Nodelet() final = default;
+  void onInit() final;
+
+private:
+  std::shared_ptr<livox_ros::DriverNode> livox_node_;
```
Comment (Roboterbastler) [inline include/livox_ros_driver2/nodelet.h:24]: Maybe also include `<memory>` for completeness...

File: nodelets.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<library path="lib/liblivox_ros_driver2_nodelet">
+    <class name="livox_ros_driver2/Nodelet"
+           type="livox_ros::Nodelet"
+           base_class_type="nodelet::Nodelet">
+      <description>
+        The ROS device driver for Livox 3D LiDARs
+      </description>
+    </class>
+  </library>
```
Comment (Roboterbastler) [inline nodelets.xml]: Indentation is displaced a bit

File: src/driver_node.h [hunk 3]
```diff
@@ -35,14 +35,15 @@ class Lddc;
...
   DriverNode() = default;
-  DriverNode(const DriverNode &) = delete;
+  DriverNode(const NodeHandle& rhs) : ros::NodeHandle(rhs) {};
   ~DriverNode();
-  DriverNode &operator=(const DriverNode &) = delete;
-
+  DriverNode &operator=(const DriverNode& rhs) = default;
   DriverNode& GetNode() noexcept;
 
+
```
Comment (Roboterbastler) [inline src/driver_node.h]: Additional newline needed?

File: package_ROS1.xml [hunk 4]
```diff
@@ -70,6 +72,8 @@
   <exec_depend>message_runtime</exec_depend>
   <exec_depend>rosbag</exec_depend>
   <exec_depend>pcl_ros</exec_depend>
+  <exec_depend>nodelet</exec_depend>
+  <exec_depend>pluginlib</exec_depend>
```
Comment (Roboterbastler) [inline package_ROS1.xml]: Could these be just `<depend>` tags?

File: package.xml [hunk 5]
```diff
@@ -70,6 +72,8 @@
   <exec_depend>message_runtime</exec_depend>
   <exec_depend>rosbag</exec_depend>
   <exec_depend>pcl_ros</exec_depend>
+  <exec_depend>nodelet</exec_depend>
+  <exec_depend>pluginlib</exec_depend>
```
Comment (Roboterbastler) [inline package.xml]: Could these be just `<depend>` tags?

**Review by Roboterbastler — CHANGES_REQUESTED** [review]

**Review by Roboterbastler — APPROVED** [review]

**Pull request merged**